src/preprocessor.rs

```diff
@@ -19,7 +19,8 @@ pub static INFO_STRING_PREFIX: &str = "dot process";
 
 pub struct GraphvizPreprocessor;
 
-pub struct Graphviz<R: GraphvizRenderer> {
+pub struct Graphviz<'a, R: GraphvizRenderer> {
```

Comment:
Personally, I think the lifetimes are acceptable if needed, but not necessary. I'd just use a `Vec<String>` and avoid the potential for needing to create long lived `Strings` for these `&str`s to reference.

Still, all that said, I think it's probably fine.

Can you fix merge conflicts and let me quickly rereview the result?

Reply:
Fixed them. I have kind of refactored the whole thing to make it cleaner, using `Vec<String>` like you recommended.